src/arch/riscv/config.cmake

```diff
@@ -67,12 +67,7 @@ elseif(KernelPTLevels EQUAL 4)
     math(EXPR KernelPaddrUserTop "1 << 56")
 endif()
 
-if(KernelRiscvExtD)
-    set(KernelRiscvExtF ON)
```

Comment:
If D extension is enabled, then the F extension is also enabled should be intentional here?

Reply:
I don't think that has any intentional effect though, since the F extension ifdefs _only_ appear as an `#elif` to the D extension ifdefs in the kernel.

Reply:
This chapter describes the standard double-precision floating-point instruction-set extension, which
is named “D” and adds double-precision floating-point computational instructions compliant with
the IEEE 754-2008 arithmetic standard. The D extension depends on the base single-precision
instruction subset F

Reply:
To me, `CONFIG_HAVE_FPU` and `CONFIG_RISCV_EXT_F` are the same, that is, they both enable the most basic FPU instruction subset. Would it be useful to instead set `KernelRiscvExtF`_only_ (and not KernelHaveFpu) when we set `KernelRiscvExtD`, to be more in line with the spec?

A comment would explaining this might suffice too.

Reply:
> Seems currently no RISC-V platform enables anything there, it's all done manually? So we should even consider sanitizing things here and abort if we detect explicit combinations that we can't make sense of. That avoids building future legacy.

I'm a bit confused and not sure what you mean by stuff being done manually. Also what would we be sanitizing?

Reply:
I could not find any place that enabled the FPU on RISC-V, so the only way how to do this at the moment is specifying this as an explicit build parameter manually. Now is `KernelRiscvExtD` is enabled there, we need some sanitizing in CMake that also enabled `KernelRiscvExtF` then if this is not set - and it might better abort is it detects this is already set explicitly to "OFF", as this looks like a broken configuration then (or something that needs to be clarified, what this is supposed to achieve)